src/main/java/org/reaktivity/nukleus/maven/plugin/internal/ast/parse/AstParser.java

```diff
@@ -585,8 +602,75 @@ private static int parseInt(
         }
     }
 
-    private static int parseInt(String text)
+    private static int parseInt(
+        String text)
     {
         return Integer.decode(text);
     }
+
+    private final class EnumVisitor extends NukleusBaseVisitor<AstNode>
+    {
+        @Override
+        public AstNode visitEnum_type(
+            Enum_typeContext ctx)
+        {
+            return super.visitEnum_type(ctx);
+        }
+
+        @Override
+        public AstNode visitUint8_type(
+            Uint8_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.UINT8);
+            return super.visitUint8_type(ctx);
+        }
+
+        @Override
+        public AstNode visitUint16_type(
+            Uint16_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.UINT16);
+            return super.visitUint16_type(ctx);
+        }
+
+        @Override
+        public AstNode visitUint32_type(
+            Uint32_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.UINT32);
+            return super.visitUint32_type(ctx);
+        }
+
+        @Override
+        public AstNode visitUint64_type(
+            Uint64_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.UINT64);
+            return super.visitUint64_type(ctx);
+        }
+
+        @Override
+        public AstNode visitString_type(
+            String_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.STRING);
+            return super.visitString_type(ctx);
+        }
+
+        @Override
+        public AstNode visitString16_type(
+            String16_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.STRING16);
+            return super.visitString16_type(ctx);
+        }
+
+        @Override
+        public AstNode visitString32_type(
+            String32_typeContext ctx)
+        {
+            enumBuilder.valueType(AstType.STRING32);
+            return super.visitString32_type(ctx);
+        }
```

Comment:
Move `visitEnum_value` here instead of in the main visitor.
Also move majority of `visitEnum_type` here, returning an `AstEnumNode.Builder` from `EnumVisitor`.

Suggest the following:
```
    public final class EnumVisitor extends NukleusBaseVisitor<AstEnumNode.Builder>
    {
        private final AstEnumNode.Builder enumBuilder;

        public EnumVisitor()
        {
            this.enumBuilder = new AstEnumNode.Builder();
        }

        @Override
        public AstEnumNode.Builder visitEnum_type(
            Enum_typeContext ctx)
        {
            enumBuilder.name(ctx.ID().getText());

            if (ctx.LEFT_BRACKET() != null)
            {	
                enumBuilder.valueType(AstType.UINT8);	
            }

            return super.visitEnum_type(ctx);
        }

	protected AstEnumNode.Builder defaultResult()
        {
		return enumBuilder;
	}
```
...plus the existing methods on `EnumVisitor`, but with return type `AstEnumNode.Builder`.

Then in main visitor, suggest the following:
```
    @Override
    public AstEnumNode visitEnum_type(
        Enum_typeContext ctx)
    {
        visitLocalName(ctx.ID().getText());

        AstEnumNode.Builder enumBuilder = new EnumVisitor().visitEnum_type(ctx);
        AstEnumNode enumeration = enumBuilder.build();

        AstScopeNode.Builder scopeBuilder = scopeBuilders.peekLast();
        if (scopeBuilder != null)
        {
            scopeBuilder.enumeration(enumeration);
        }

        return enumeration;
    }
```